src/OrleansRuntime/Streams/PersistentStream/PersistentStreamPullingAgent.cs

```diff
@@ -555,17 +563,8 @@ private void StartInactiveCursors(StreamConsumerCollection streamData)
                 }
             }
             StreamHandshakeToken newToken = await batchDeliveryTask;
-            if (newToken != null)
-            {
-                consumerData.LastToken = newToken;
-                consumerData.Cursor = queueCache.GetCacheCursor(consumerData.StreamId.Guid,
-                    consumerData.StreamId.Namespace, newToken.Token);
-            }
-            else
-            {
-                consumerData.LastToken = StreamHandshakeToken.CreateDeliveyToken(batch.SequenceToken); // this is the currently delivered token
-            }
-
+            consumerData.LastToken = StreamHandshakeToken.CreateDeliveyToken(batch.SequenceToken); // this is the currently delivered token
```

Comment:
It does not matter and code is simpler this way.  Think of it like this:
1) We deliver the event.
2) When the delivery is completed, we update the last token.
3) If a correction was returned (newToken != null), then we update the last token from the correction and get a new cursor.
This cleanly separates the delivery functionality for the handling of the response.